partiql-planner/src/testFixtures/resources/inputs/basics/case.sql

```diff
@@ -292,6 +292,56 @@ CASE t_item.t_string
     ELSE t_item.t_any
 END;
 
+--#[case-when-35]
+-- type: decimal(10,5)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE null
+END;
+
+--#[case-when-36]
+-- type: decimal(10,5)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE t_item.t_decimal_5_3
+END;
+
+--#[case-when-37]
+-- type: varchar(10)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_varchar_10
+    ELSE null
+END;
+
+--#[case-when-38]
+-- type: varchar(10)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_varchar_10
+    ELSE t_item.t_varchar_5
+END;
+
+--#[case-when-39]
+-- type: char(10)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE null
+END;
+
+--#[case-when-40]
+-- type: char(10)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE t_item.t_char_5
+END;
+
+--#[case-when-41]
+-- type: varchar(10)
+CASE t_item.t_string
```

Comment:
Could add a few more tests
- Mixing arbitrary decimal with parameterized decimals
- Mixing parameterized decimal with other numeric types
- Using a decimal with a specified `precision` and unspecified `scale` 
- Mixing arbitrary varchar / char with parameterized varchar / char

Reply:
Also, perhaps it would be helpful to add a `toString()` to static type's `DecimalType` since it currently does not output the precision/scale constraint.

Reply:
Great suggestion on test addition. 

> Using a decimal with a specified precision and unspecified scale
Unspecified scale essentially just default to 0. In my opinion this is perhaps better covered by syntax to IR. 

>it would be helpful to add a toString() to static type's DecimalType...
Yes. I agree. But perhaps leaving this out of this PR?

Reply:
> Yes. I agree. But perhaps leaving this out of this PR?

I'm fine leaving it out or adding it. Just may be easier to debug anywhere we print types in the error message. Also applies to the string constraints.